### PR TITLE
PYTHON-4614 Do not test PyPy with OpenSSL 1.0.2

### PR DESCRIPTION
**Pull request by blink1073**
Since we're using hatch for testing, and hatch requires cryptography.

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
cryptography builds wheels for pypy: https://pypi.org/project/cryptography/#files

Why are we installing from one of those wheels?

**Comment by blink1073** [comment]
> Why aren't we installing from one of those wheels?

They require newer versions of gcc.

File: .evergreen/combine-coverage.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ fi
 createvirtualenv "$PYTHON_BINARY" covenv
 # Keep in sync with run-tests.sh
 # coverage >=5 is needed for relative_files=true.
-pip install -q "coverage>=5,<=7.5"
+pip install -q "coverage[toml]>=5,<=7.5"
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/combine-coverage.sh:18]: It's needed for `pyproject.toml` support in `coverage`, I think we had been picking `tomlib` from other places, but this makes it explicit.

File: .evergreen/utils.sh [hunk 2]
```diff
@@ -67,7 +67,6 @@ createvirtualenv () {
 
     export PIP_QUIET=1
     python -m pip install --upgrade pip
-    python -m pip install --upgrade hatch
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/utils.sh:70]: It is installed in `hatch.sh`.  It didn't make sense to install it here as well.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**